src/core/agent/services/keriaNotificationService.ts

```diff
@@ -0,0 +1,743 @@
+import { OperationPendingRecordType } from "../records/operationPendingRecord.type";
+import { OperationPendingRecord } from "../records/operationPendingRecord";
+import { IonicStorage } from "../../storage/ionicStorage";
+import { ConnectionHistoryType } from "./connection.types";
+import { MultiSigService } from "./multiSigService";
+import { IpexCommunicationService } from "./ipexCommunicationService";
+import { IdentifierService } from "./identifierService";
+
+class KeriaNotificationService extends AgentService {
+  static readonly NOTIFICATION_NOT_FOUND = "Notification record not found";
+  static readonly POLL_KERIA_INTERVAL = 2000;
+  static readonly CHECK_READINESS_INTERNAL = 25;
+
+  protected readonly notificationStorage!: NotificationStorage;
+  protected readonly identifierStorage: IdentifierStorage;
+  protected readonly operationPendingStorage: OperationPendingStorage;
+  protected readonly connectionStorage: ConnectionStorage;
+  protected readonly ipexMessageStorage: IpexMessageStorage;
+  protected readonly credentialStorage: CredentialStorage;
+  protected readonly basicStorage: BasicStorage;
+  protected readonly multiSigs: MultiSigService;
+  protected readonly ipexCommunications: IpexCommunicationService;
+  protected readonly identifiers: IdentifierService;
+  protected readonly getKeriaOnlineStatus: () => boolean;
+  protected readonly markAgentStatus: (online: boolean) => void;
+  protected readonly connect: (retryInterval?: number) => Promise<void>;
+
+  protected pendingOperations: OperationPendingRecord[] = [];
+  private loggedIn = true;
+
+  constructor(
+    agentServiceProps: AgentServicesProps,
+    notificationStorage: NotificationStorage,
+    identifierStorage: IdentifierStorage,
+    operationPendingStorage: OperationPendingStorage,
+    connectionStorage: ConnectionStorage,
+    ipexMessageStorage: IpexMessageStorage,
+    credentialStorage: CredentialStorage,
+    basicStorage: BasicStorage,
+    multiSigs: MultiSigService,
+    ipexCommunications: IpexCommunicationService,
+    identifiers: IdentifierService,
+    getKeriaOnlineStatus: () => boolean,
+    markAgentStatus: (online: boolean) => void,
+    connect: (retryInterval?: number) => Promise<void>
+  ) {
+    super(agentServiceProps);
+    this.notificationStorage = notificationStorage;
+    this.identifierStorage = identifierStorage;
+    this.operationPendingStorage = operationPendingStorage;
+    this.connectionStorage = connectionStorage;
+    this.ipexMessageStorage = ipexMessageStorage;
+    this.credentialStorage = credentialStorage;
+    this.basicStorage = basicStorage;
+    this.multiSigs = multiSigs;
+    this.ipexCommunications = ipexCommunications;
+    this.identifiers = identifiers;
+    this.getKeriaOnlineStatus = getKeriaOnlineStatus;
+    this.markAgentStatus = markAgentStatus;
+    this.connect = connect;
+  }
+
+  async pollNotificationsWithCb(callback: (event: KeriaNotification) => void) {
+    try {
```

Comment:
So I wanted to create 2 functions, so we could avoid having a big try catch that indents this big file. This is the first acceptance criteria

So we have `pollNotificationsWithCb` with the try catch and a `private pollNotifications` that it calls. This would be more readable